changelog/3317.feature

```diff
@@ -0,0 +1 @@
+introduce correct per node mark handling and deprecate the always incorrect existing mark handling
```

Comment:
Please add a note to the docs in `markers.rst` explaining this change and how it affects end users in detail, and post a link to that section here; I think one of the most important things we need to do about this change is to properly communicate what they are, what it solves and how it might affect users. 👍 

```rst
Introduce correct per-node mark handling, which aims to fix long-standing problems on how marks are used and propagated through objects. For more information please see `this section in the docs <some url>`_.
```